src/services/BaseService.ts

```diff
@@ -33,4 +36,32 @@ export class BaseService {
       headers,
     });
   }
+
+  /**
+   * Makes an authenticated request to the server, with the token saved in localStorage.
+   * If no token is saved throws an error.
+   * @param method Request method
+   * @param endpoint Request endpoint
+   * @param body Request body
+   * @returns Response from requst
+   */
+  public static async authenticatedRequest(
+    method: MethodType,
+    endpoint: string,
+    body?: Record<string, unknown>
+  ): Promise<Response> {
+    const token = StorageService.getToken();
+    if (!token) {
+      throw new AppError('AUTH_TOKEN_INVALID');
```

Comment:
This should be changed to a new Error such as 'AUTH_TOKEN_MISSING' as @an-p  mentioned

Reply:
Maybe `AUTH _USER_NOT_LOGGED_IN`  might be more accurate 🤔